src/Workleap.OpenApi.MSBuild/SpectralManager.cs

```diff
@@ -0,0 +1,140 @@
+using System.Runtime.InteropServices;
+using Microsoft.Build.Framework;
+using Workleap.OpenApi.MSBuild.Exceptions;
+
+namespace Workleap.OpenApi.MSBuild;
+
+internal sealed class SpectralManager : ISpectralManager
+{
+    private const string SpectralVersion = "6.11.0";
+
+    private readonly IProcessWrapper _processWrapper;
+    private readonly ILoggerWrapper _loggerWrapper;
+    private readonly string _openApiToolsDirectoryPath;
+    private readonly string _toolDirectory;
+
+    public SpectralManager(IProcessWrapper processWrapper, ILoggerWrapper loggerWrapper, string openApiToolsDirectoryPath)
+    {
+        this._processWrapper = processWrapper;
+        this._loggerWrapper = loggerWrapper;
+        this._openApiToolsDirectoryPath = openApiToolsDirectoryPath;
+        this._toolDirectory = Path.Combine(openApiToolsDirectoryPath, $"spectral/{SpectralVersion}");
+    }
+
+    public async Task InstallSpectralAsync(CancellationToken cancellationToken)
+    {
+        this.CreateRequiredDirectories();
+        var executableFileName = this.GetSpectralFileName();
+
+        var url = $"https://github.com/stoplightio/spectral/releases/download/v{SpectralVersion}/{executableFileName}";
+
+        await this.DownloadFileAsync(url, $"{this._toolDirectory}/{executableFileName}", cancellationToken);
+    }
+
+    private void CreateRequiredDirectories()
+    {
+        Directory.CreateDirectory(Path.Combine(this._openApiToolsDirectoryPath, "spectral"));
```

Comment:
Unecessary as already implemented with `Directory.CreateDirectory(this._toolDirectory);`. Because of that `this._openApiToolsDirectoryPath` becomes useless